backend/presenters/dataset.py

```diff
@@ -32,7 +32,8 @@ async def save_comparison(data):
 
 async def generate_dataset(file: UploadFile):
     calculator = VarianceCalculator()
-    dataset_file = CSVFile(file.file, calculator)
+    dataset_file = CSVFile(file.file)
+    calculator.process_dataset(dataset_file)
```

Comment:
Good idea for isolating the dataset processing into only the bias calculator class instead of having it in the csv file and dataset file classes.